src/cursor/abstract_cursor.ts

```diff
@@ -767,15 +781,26 @@ export abstract class AbstractCursor<
         'Unexpected null selectedServer. A cursor creating command should have set this'
       );
     }
+    const getMoreOptions = {
+      ...this.cursorOptions,
+      session: this.cursorSession,
+      batchSize,
+      omitMaxTimeMS: this.cursorOptions.omitMaxTimeMS
+    };
+
+    if (
+      this.cursorOptions.tailable &&
+      this.cursorOptions.awaitData &&
+      this.cursorOptions.maxAwaitTimeMS != null
+    ) {
+      getMoreOptions.maxTimeMS = this.cursorOptions.maxAwaitTimeMS;
+    }
+
```

Comment:
thoughts on just calculating omitMaxTimeMS inline and not storing it on the cursor's options?
```typescript
const getMoreOptions = {
	...this.cursorOptions,
	session: this.cursorSession,
	batchSize,
};

if (timeoutMS != null) {
	if (...) options.omitMaxTimeMS = true;
}
```

Reply:
Wouldn't we still need to calculate it for the initial command? It works better in my opinion to continue to have it on the internal cursor options since it lets us just spread the options  wherever needed across all of our cursor subclasses. I can change the getMore code here to just use the value obtained by the spread though. Also instead of recalculating it every time we enter `fetchBatch` as we do currently, we could just set it to the value needed after `_initialize` is called in `cursorInit`

Reply:
Yeah, I'm suggesting we just calculate it for the initial cursor command inline (in `_initialize()`).

Here are my thoughts - if you disagree, that's okay and we can resolve this conversation.  This is non-blocking.

`omitMaxTimeMS` isn't a cursor option - it's an option provided to operation execution that cursors calculate given  cursor options and state.  We have a history of attaching data on large objects because its convenient, but I don't think that's the right approach because it leads to a blurring of the boundaries between the data our various objects contain (for example, I think there's a reason we separate out BSON options with pluckBSONSerializationOptions - we have all our options on a shared object and need to separate them out with functions instead of just separating them out when constructing the options object).

I'm also a proponent of keeping data as closely scoped to where its needed as possible. `omitMaxTimeMS` is only needed in exactly two places: when constructing the initialization options and when constructing the getMore options.  It is a little strange to calculate the value of an option in the constructor that only gets used in `_initialize()`.  

Additionally, `omitMaxTimeMS` isn't used in any subclasses and I don't know of any CSOT requirement that will require this in the future.  Storing it in a protected property leaks implementation details of the abstract cursor.

I see the point about convenience about spreading cursor options, but is it that much simpler than something like this?:

```typescript
_initialize() {
  const options = { 
    ...this.cursorOptions, 
    omitMaxTimeMS: this.timeoutMS != null && ... 
  }
}

getMore() {
  const options = { 
    ...this.cursorOptions, 
    omitMaxTimeMS: this.timeoutMS != null 
  }
}
```